src/librustc_mir/transform/type_check.rs

```diff
@@ -1210,45 +1210,13 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
     ) {
         let tcx = self.tcx();
 
-        self.prove_aggregate_predicates(aggregate_kind, location);
-
-        match aggregate_kind {
+        if *aggregate_kind == AggregateKind::Tuple {
```

Comment:
Nit: if we are going to "return early" for tuples, we should have `bug!` cases for the `prove_aggregate_predicates` function etc. Otherwise if people tried to add code there there might be confusion.

Reply:
The other thing we could do is to leave prove_aggregate_predicates before the early return as it was ... so if you want to handle Tuples there you can.

Reply:
Either would be fine.

Reply:
Done!